### PR TITLE
test: add test for test coverage

### PR DESCRIPTION
**Pull request by lucasfang**
### Purpose

Linked issue: close #xxx

This change adds targeted unit tests that close concrete coverage gaps identified from the
LCOV report. It is test-only: no production code is modified. Every case locks down a real
behavioral contract, boundary condition, or previously-uncovered error/default branch rather
than repeating an already-covered happy path.

The added tests fall into two areas:

- **Common utilities** — error/stringification and failure paths that upper layers rely on:
  - `Status::CodeAsString` mapping for every `StatusCode` (including the out-of-range default
    and the instance overload), and `Status::Abort` fatal-exit behavior.
  - `FileTypeUtils::ToString` mapping, plus a `.tmp` unwrap boundary that must *not* unwrap.
  - Arrow memory-pool adaptor out-of-memory conversion for both `Allocate` and `Reallocate`,
    covering the underlying pool returning `nullptr` and throwing `std::bad_alloc`.
  - Logging: full severity mapping (including the default branch), glog file-sink write-to-disk,
    and custom logger-creator registration.
  - Block compression factory creation from `CompressOptions` (case-insensitive codec names,
    unsupported name rejected) and from `BlockCompressionType` (declared-but-unimplemented `LZO`
    hits the default branch).

- **Variant subsystem** — boundary and error paths of type access, JSON rendering, shredding
  schema inference, shredded write/reassembly round-trips, and the shredding read-plan factory:
  - `GenericVariant` type-info accessor, type-mismatch rejection of the typed getters, and the
    primitive width branches of `ValueSize`.
  - `VariantJsonUtils` control-char escaping, negative-year dates, timestamp fraction/floor-div
    edges, and fixed zone-offset parsing (valid and invalid forms).
  - `variant_get` casts: UUID-to-string only, Java-style float/double stringification, date
    source, missing cast executor, all scalar Arrow builders, and object-to-list rejection.
  - `InferVariantShreddingSchema` object/array merges, variant-null handling, depth limit,
    scalar leaf types, long/decimal merge widening and overflow fallback, and cardinality drop.
  - `VariantAccessUtils` access-spec parsing (malformed descriptions rejected) and file-field
    clipping (unprunable/unshredded/missing-metadata/narrow-to-requested-keys).
  - `VariantShreddingReadPlanFactory` full-variant reassembly, unshredded value fallback, null
    metadata, shredded typed extraction, nested-variant rebuild with type-mismatch, and
    schema-evolution column absence.
  - `variant_shredding` round-trips for byte/short/float/binary/date typed columns, timestamp
    schema parsing and reassembly, `ScalarSchemaToArrowType`, and invalid-schema rejection.

### Tests

- `src/paimon/common/utils/status_test.cpp`: `TestCodeAsString`, `StatusDeathTest.TestAbort`.
- `src/paimon/common/utils/file_type_test.cpp`: `TestToString`; new `.tmp` boundary case in
  `TestInvalidTempWrapperFallsBackToOriginalName`.
- `src/paimon/common/utils/arrow/mem_utils_test.cpp`: `TestAllocateOutOfMemory`,
  `TestReallocateOutOfMemory` (with a `FailingMemoryPool` helper).
- `src/paimon/common/logging/logging_test.cpp`: `TestLogAllSeverities`,
  `TestGlogWritesLogFileToDisk`, `TestRegisterCustomLoggerCreator`.
- `src/paimon/common/compression/block_compression_factory_test.cpp`:
  `TestCreateFromCompressOptions`, `TestCreateFromCompressionType`.
- `src/paimon/common/data/variant/generic_variant_test.cpp`: `GetTypeInfoReturnsHeaderBits`,
  `TypedGettersRejectMismatchedTypes`, `ValueSizeCoversAllPrimitiveWidths`.
- `src/paimon/common/data/variant/variant_json_utils_test.cpp`: `AppendEscapedJsonControlChars`,
  `DateToStringNegativeYear`, `TimestampToStringEdgeCases`, `ZoneOffsetParsing`.
- `src/paimon/common/data/variant/variant_get_test.cpp`: `UuidSourceCastsToStringOnly`,
  `FloatingPointToStringMatchesJava`, `DateSource`, `MissingCastExecutorYieldsNull`,
  `ScalarArrowBuilders`, `ListFromNonArrayIsNull`.
- `src/paimon/common/data/variant/infer_variant_shredding_schema_test.cpp`:
  `MergeObjectsWithDisjointFields`, `VariantNullSamplesAndFields`, `ArraysMerge`,
  `ArrayBeyondDepthLimitStaysVariant`, `ScalarLeafTypes`, `LargeIntegerAndDecimalMerging`,
  `LongMergedWithIntegralDecimalStaysLong`, `SmallFractionalDecimalPrecisionAdjusted`,
  `DecimalMergeOverflowFallsToVariant`, `ObjectWithAllRareFieldsStaysUnshredded`.
- `src/paimon/common/data/variant/variant_access_utils_test.cpp` (new file):
  `ParseAccessSpecsRejectsNonProjection`, `ParseAccessSpecsRejectsMalformedDescription`,
  `ParseAccessSpecsRoundTripsBuildMetadata`, `ClipRejectsNonStructFileField`,
  `ClipUnprunablePathsReturnFileUnchanged`, `ClipUnshreddedFileFieldReturnedUnchanged`,
  `ClipMissingMetadataFails`, `ClipNarrowsToRequestedKeys`.
- `src/paimon/common/data/variant/variant_shredding_read_plan_factory_test.cpp` (new file):
  `FullVariantReadOfShreddedFile`, `AccessProjectionOnUnshreddedFile`,
  `AccessProjectionRejectsNullMetadata`, `AccessProjectionOnShreddedFile`,
  `NestedVariantColumnAndTypeMismatch`, `ColumnAbsentInFileSkipped`.
- `src/paimon/common/data/variant/variant_shredding_test.cpp`: `ScalarShreddingIntegerWidths`,
  `ScalarShreddingFloatAndBinary`, `ScalarShreddingDate`, `TimestampSchemaParsing`,
  `TimestampReassembly`, `ScalarSchemaToArrowType`, `InvalidShreddingSchemas`.

### API and Format

No API, storage format, or protocol change. This PR only adds tests (and one CMake test-source
registration); no headers under `include/` and no production sources are modified.

### Documentation

No new feature.

### Generative AI tooling

Generated-by: Qoder

### CODE REVIEW

File: src/paimon/common/logging/logging_test.cpp [hunk 1]
```diff
@@ -36,4 +82,99 @@ TEST(LoggerTest, TestMultiThreadGetLogger) {
...
+    const bool prev_logtostderr = FLAGS_logtostderr;
+    const bool prev_timestamp_in_name = FLAGS_timestamp_in_logfile_name;
+    const int32_t prev_minloglevel = FLAGS_minloglevel;
+
+    // A unique, empty directory so the only file inside is the one glog creates for us.
+    fs::path log_dir = fs::temp_directory_path() /
+                       ("paimon_glog_disk_" + std::to_string(RandomNumber(0, 1'000'000'000)));
+    fs::create_directories(log_dir);
+    const fs::path base = log_dir / "paimon_demo";
+
```
Comment (lxy-9602) [inline src/paimon/common/logging/logging_test.cpp:114]: Why not use `UniqueTestDirectory` and paimon fs?

**Review by lxy-9602 — APPROVED** [review]
+1
